src/main/scala-templates/limited/com/normation/plugins/datasources/EnablePluginImpl.scala

```diff
@@ -0,0 +1,99 @@
+/*
+*************************************************************************************
+* Copyright 2017 Normation SAS
+*************************************************************************************
+*
+* This file is part of Rudder.
+*
+* Rudder is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* In accordance with the terms of section 7 (7. Additional Terms.) of
+* the GNU General Public License version 3, the copyright holders add
+* the following Additional permissions:
+* Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+* Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+* Public License version 3, when you create a Related Module, this
+* Related Module is not considered as a part of the work and may be
+* distributed under the license agreement of your choice.
+* A "Related Module" means a set of sources files including their
+* documentation that, without modification of the Source Code, enables
+* supplementary functions or services in addition to those offered by
+* the Software.
+*
+* Rudder is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+*
+*************************************************************************************
+*/
+
+package com.normation.plugins.datasources
+
+import com.normation.license._
+import java.util.Formatter.DateTime
+import java.io.InputStreamReader
+import org.joda.time.DateTime
+import bootstrap.rudder.plugin.CheckRudderPluginDatasourcesEnable
+
+
+/*
+ * This template file will processed at build time to choose
+ * the correct immplementation to use for the interface.
+ * The default implementation is to always enable status.
+ *
+ * The class will be loaded by ServiceLoader, it needs an empty constructor.
+ */
+
+final class CheckRudderPluginDatasourcesEnableImpl() extends CheckRudderPluginDatasourcesEnable {
+  // here are processed variables
+  val CLASSPATH_KEYFILE = "license.pubkey" // "${plugin-resource-publickey}"
+  val FS_SIGNED_LICENSE = "/home/fanf/Rudder/license/license.sign"  // "${plugin-resource-license}"
```

Comment:
How will this variables be filled with correct value ?? I Think they will be injected at build time

Reply:
Yes, see the makefile. We are filtering the source file (that's why it is in template, then go to target/generated, then compiled) and passing the value as -D param in mvn command line